v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/freecollateral/DydxReceiptFreeCollateralViewModel.kt

```diff
@@ -2,113 +2,68 @@ package exchange.dydx.trading.feature.receipt.components.buyingpower
 
 import androidx.lifecycle.ViewModel
 import dagger.hilt.android.lifecycle.HiltViewModel
-import exchange.dydx.abacus.output.Subaccount
+import exchange.dydx.abacus.output.SubaccountPendingPosition
 import exchange.dydx.abacus.output.SubaccountPosition
-import exchange.dydx.abacus.output.input.TradeInput
-import exchange.dydx.abacus.output.input.TransferInput
 import exchange.dydx.abacus.protocols.LocalizerProtocol
 import exchange.dydx.dydxstatemanager.AbacusStateManagerProtocol
 import exchange.dydx.trading.common.DydxViewModel
 import exchange.dydx.trading.common.formatter.DydxFormatter
-import exchange.dydx.trading.feature.receipt.ReceiptType
 import exchange.dydx.trading.feature.shared.views.AmountText
-import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filterNotNull
 import kotlinx.coroutines.flow.flatMapLatest
-import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
 import javax.inject.Inject
 
 @HiltViewModel
 class DydxReceiptFreeCollateralViewModel @Inject constructor(
     private val localizer: LocalizerProtocol,
     private val abacusStateManager: AbacusStateManagerProtocol,
     private val formatter: DydxFormatter,
-    private val receiptTypeFlow: Flow<@JvmSuppressWildcards ReceiptType?>,
 ) : ViewModel(), DydxViewModel {
 
-    @OptIn(ExperimentalCoroutinesApi::class)
     val state: Flow<DydxReceiptFreeCollateralView.ViewState?> =
-        receiptTypeFlow
-            .flatMapLatest { receiptType ->
-                when (receiptType) {
-                    is ReceiptType.Trade -> {
-                        combine(
-                            abacusStateManager.state.selectedSubaccountPositions,
-                            abacusStateManager.state.tradeInput,
-                        ) { positions, tradeInput ->
-                            createViewState(positions, tradeInput)
-                        }
-                    }
-                    is ReceiptType.Transfer -> {
-                        combine(
-                            abacusStateManager.state.selectedSubaccount,
-                            abacusStateManager.state.transferInput,
-                        ) { subaccount, transferInput ->
-                            createViewState(subaccount, transferInput)
-                        }
-                    }
-                    else -> flowOf()
+        abacusStateManager.marketId
+            .filterNotNull()
+            .flatMapLatest { marketId ->
+                combine(
+                    abacusStateManager.state.selectedSubaccountPositionOfMarket(marketId),
+                    abacusStateManager.state.selectedSubaccountPendingPositions.map { it?.firstOrNull { it.marketId == marketId } },
+                ) { position, pendingPosition ->
+                    Pair(position, pendingPosition)
```

Comment:
nit: think we can just call `createViewState()` directly here.

also kotlin tip: `to` is a built in `infix fun` that creates Pairs. ex: `position to pendingPosition`

Reply:
Updated